### PR TITLE
Onboarding refactoring

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?
- Fixes some small bugs
  - Modal skip prevents onboard from displaying
  - Sometimes the confirm msg bugged out a bit
- Refactors onboarding js to break it up a bit

I've tested 
- No modal
- Competition modal
- School finder modal

For modals I tried the main submit button, and any skip/no thanks options.

In this PR it works for every case described above ^
#### How should this be reviewed?

Test cases above ^
#### Any background context you want to provide?

what else?
#### Relevant tickets

Fixes #
#### Checklist
- [ ] Tested on staging.

### CODE REVIEW

File: lib/themes/dosomething/paraneue_dosomething/js/components/InstructionSlide.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ class InstructionSlide extends React.Component {
 
   render() {
     const competition = Drupal.settings.dosomethingSignup ? Drupal.settings.dosomethingSignup.competition : null;
+    const display_competition_message = competition && competition.joined && competition.messages;
```
Comment (DFurnes) [inline lib/themes/dosomething/paraneue_dosomething/js/components/InstructionSlide.js]: This should be camel-case. 🐫

Comment (itsjoekent) [inline lib/themes/dosomething/paraneue_dosomething/js/components/InstructionSlide.js]: _whoops_

**Comment by DFurnes** [comment]
🚢

**Comment by itsjoekent** [comment]
🚂 🚋 🚋

**Pull request merged**